src/main/java/net/torocraft/minecoprocessors/processor/Register.java

```diff
@@ -52,5 +52,15 @@
    *
    * <ul> <li><b>bit 0:</b> front port</li> <li><b>bit 1:</b> back port</li> <li><b>bit 2:</b> left port</li> <li><b>bit 3:</b> right port</li> </ul>
    */
-  ADC
+  ADC,
+  /**
+   * Digital Input Data Register (reads all input ports as nibble).
+   * Bit order: (pf<<0)|(pb<<1)|(pl<<2)|(pr<<3)
+   */
+  IDR,
```

Comment:
Interesting idea.  How do you handle conflicts between IDR or ODR and PF/PR/PL/PB?

Reply:
Hi, based on the assumption of the one-instruction-per-tick I synchronise IDR while reading the inputs (`updateInputPorts()`), and ODR while writing the outputs (`updateOutputs()`). pf,pb, etc have an edge detection to firing `neighbourChanged` on adjacent blocks only when appropriate, and this was done for the ODR, too. So for one tick, only one of `odr` or ports should change if I did not manage to build in a late night bug ;). Cheers,-